src/components/ProfileMenu.tsx

```diff
@@ -0,0 +1,101 @@
+import React from 'react';
+import {
+  Menu,
+  MenuButton,
+  IconButton,
+  MenuList,
+  Text,
+  Avatar,
+  Button,
+  useColorModeValue,
+} from '@chakra-ui/react';
+import { signOut } from 'next-auth/react';
+
+// Helper to color-code the "X/Y Absences Taken"
+function getAbsenceColor(usedAbsences: number) {
+  if (usedAbsences <= 6) return 'green.500';
+  if (usedAbsences <= 8) return 'yellow.500';
+  return 'red.500';
+}
+
+// Same interface as used in CalendarHeader, or define your own
+interface UserData {
+  name: string;
+  email: string;
+  image?: string;
+  usedAbsences: number;
+  numOfAbsences: number;
+}
+
+interface ProfileMenuProps {
+  userData?: UserData;
+}
+
+const ProfileMenu: React.FC<ProfileMenuProps> = ({ userData }) => {
+  const absenceColor = userData
+    ? getAbsenceColor(userData.usedAbsences)
+    : 'gray.500';
+
+  return (
+    <Menu>
+      <MenuButton
+        as={IconButton}
+        icon={
+          <Avatar
+            name={userData?.name ?? 'User'}
+            src={userData?.image}
```

Comment:
Hmm, not sure why but sometimes I need to navigate to the /profile page, and then navigate back to the calendar to get the image to show up. I assume it works on the profile page since it uses a use effect to make sure the data is available whereas this does not? Not quite sure, but would be nice if the image / other data was useeffected for this component.
<img width="225" alt="image" src="https://github.com/user-attachments/assets/fd45d2db-2c5c-4920-8043-bc1348f4d3e8" />

Reply:
Should be:
<img width="232" alt="image" src="https://github.com/user-attachments/assets/d732639b-19df-4d9c-8d7f-49d114aaeadb" />